R/readWrite.R

```diff
@@ -0,0 +1,96 @@
+#' Outputs a GTF file, transcript-count file, and gene-count file from bambu
+#' @title transcript to gene expression
+#' @param se a summarizedExperiment object from \code{\link{bambu}}
+#' @param path the destination of the output files (gtf, transcript counts, and gene counts)
+#' @return gtf a GTF dataframe
+#' @export
+write.bambu <- function(transcript_se,path){
+  if (missing(transcript_se) | missing(path)){
+    stop('Both summarizedExperiment object from bambu and the path for the output files are required.')
+  }else{
+    dir.create(path)
+    transcript_grList <- rowRanges(transcript_se)
+    transcript_gtffn <- paste(path,"bambu_transcript_exon.gtf",sep="/")
+    transcript_geneIDs <-as.data.frame(rowData(transcript_se))[,c(1,2)]
+    gtf <- write.gtf(grList=transcript_grList,file=transcript_gtffn,geneIDs=transcript_geneIDs)
+    transcript_counts <- as.data.frame(assays(transcript_se)$counts)
+    transcript_countsfn <- paste(path,"counts_transcript.txt",sep="/")
+    write.table(transcript_counts, file= transcript_countsfn, sep="\t",quote=FALSE)
+    gene_se <- transcriptToGeneExpression(se)
+    gene_counts <- as.data.frame(assays(gene_se)$counts)
+    gene_countsfn <- paste(path,"counts_gene.txt",sep="/")
+    write.table(gene_counts, file= gene_countsfn, sep="\t", quote=FALSE)
+    }
+  return (gtf)
+}
+#' Outputs a GTF file for the nanorna-bam nextflow pipeline 
+#' @title transcript to gene expression
+#' @param grList a GRangesList object
+#' @param file the output gtf file name
+#' @param geneIDs an optional dataframe of geneIDs (column 2) with the corresponding transcriptIDs (column 1)
+#' @return gtf a GTF dataframe
+#' @export
+write.gtf <- function (grList,file,geneIDs) {
```

Comment:
to be consistent with bambu, the parameters could be (annotation, file, geneIDs = NULL). When annotation was created the geneIDs are optional as they are stored in mcols(annotation)$GENEID (can be tested with !is.null(mcols(annotation, use.names=FALSE)$GENEID))

Reply:
The parameters are now (annotation, file, geneIDs = NULL). Now the write.gtf function will output gtf files with both gene and transcript IDs even if geneIDs  = NULL. The write.bambu's write.gtf will also not be taking geneIDs to generate a gtf file.